src/mpi_t/events_impl.c

```diff
@@ -121,6 +126,7 @@ void MPIR_T_event_instance(int event_index, MPI_T_cb_safety cb_safety, void *dat
             event_reg->dropped_count++;
         }
     }
+#endif /* HAVE_MPIT_EVENTS */
```

Comment:
Now that `MPIR_T_DO_EVENT` takes care of the turning off events, is it still necessary to comment out the body of these functions?

Reply:
Event_instance doesn't need to be. For the internal registration functions, I suppose I can add macro wrappers for them, too. That way the function can exist normally, but no events or sources will be reported when disabled.

Reply:
I was thinking that leaving the registration active would be harmless, but I guess that may confuse the user. I guess we can just leave it as is for now.

Reply:
If we report events and sources, the user may be confused why the callbacks never execute. On the PVAR side, we report no PVARs available unless explicitly enabled.

Reply:
Added wrappers for all internal APIs and removed `#ifdefs` from within the functions.

Reply:
Hmm, so turning the registration functions into empty macros causes warnings for unused variables:
```
In directory: /var/lib/jenkins-slave/workspace/mpich-warnings/compiler/gcc-4/config/ch4-ucx/label/centos64_review
  CC       src/mpid/ch4/netmod/ucx/lib_libmpi_la-ucx_init.lo
src/mpid/ch4/src/ch4r_recvq.c: In function ‘MPIDIG_recvq_init’:
src/mpid/ch4/src/ch4r_recvq.c:45:14: warning: unused variable ‘array_of_displacements’ [-Wunused-variable]
     MPI_Aint array_of_displacements[1] = { 0 };
              ^
src/mpid/ch4/src/ch4r_recvq.c:44:18: warning: unused variable ‘array_of_datatypes’ [-Wunused-variable]
     MPI_Datatype array_of_datatypes[1] = { MPI_INT };
```
I'll revert those wrappers for now.